repos/system_upgrade/common/actors/targetuserspacecreator/libraries/userspacegen.py

```diff
@@ -557,22 +564,71 @@ def _get_rhui_available_repoids(context, cloud_repo):
     return set(repoids)
 
 
+def get_copy_location_from_copy_in_task(context, copy_task):
+    basename = os.path.basename(copy_task.src)
+    dest_in_container = context.full_path(copy_task.dst)
+    if os.path.isdir(dest_in_container):
+        return os.path.join(copy_task.dst, basename)
+    return copy_task.dst
+
+
 def _get_rh_available_repoids(context, indata):
     """
     RH repositories are provided either by RHSM or are stored in the expected repo file provided by
     RHUI special packages (every cloud provider has itw own rpm).
     """
 
-    upg_path = rhui.get_upg_path()
-
     rh_repoids = _get_rhsm_available_repoids(context)
 
+    # If we are upgrading a RHUI system, check what repositories are provided by the (already installed) target clients
     if indata and indata.rhui_info:
-        cloud_repo = os.path.join(
-            '/etc/yum.repos.d/', rhui.RHUI_CLOUD_MAP[upg_path][indata.rhui_info.provider]['leapp_pkg_repo']
+        files_provided_by_clients = _query_rpm_for_pkg_files(context, indata.rhui_info.target_client_pkg_names)
+
+        def is_repofile(path):
+            return os.path.dirname(path) == '/etc/yum.repos.d' and os.path.basename(path).endswith('.repo')
+
+        def extract_repoid_from_line(line):
+            return line.split(':', maxsplit=1)[1].strip()
```

Comment:
most probably stupid evening question (I have yet to test it) but why do we split on the colon here?

Reply:
Because the output looks like `<field>: value` e.g  `repoid: myrepo`